examples/TestEm3/electrons.cu

```diff
@@ -247,18 +258,86 @@ static __device__ __forceinline__ void TransportElectrons(Track *electrons, cons
       continue;
     }
 
-    // Perform the discrete interaction, make sure the branched RNG state is
-    // ready to be used.
-    newRNG.Advance();
     // Also advance the current RNG state to provide a fresh round of random
     // numbers after MSC used up a fair share for sampling the displacement.
     currentTrack.rngState.Advance();
 
+    if constexpr (IsElectron) g_nextInteractionForEl[i] = winnerProcessIndex;
+    else g_nextInteractionForPos[i] = winnerProcessIndex;
+    assert(i < (IsElectron ?
+        sizeof(g_nextInteractionForEl)/sizeof(g_nextInteractionForEl[0])
+      : sizeof(g_nextInteractionForPos)/sizeof(g_nextInteractionForPos[0]) ));
+  }
+}
+
+
+// Instantiate kernels for electrons and positrons.
+__global__ __launch_bounds__(ThreadsPerBlock, MinBlocksPerSM)
+void TransportElectrons(Track *electrons, const adept::MParray *active, Secondaries secondaries,
+                        adept::MParray *activeQueue, GlobalScoring *globalScoring,
+                        ScoringPerVolume *scoringPerVolume)
+{
+  TransportElectrons</*IsElectron*/ true>(electrons, active, secondaries, activeQueue, globalScoring, scoringPerVolume);
+}
+__global__ __launch_bounds__(ThreadsPerBlock, MinBlocksPerSM)
+void TransportPositrons(Track *positrons, const adept::MParray *active, Secondaries secondaries,
+                        adept::MParray *activeQueue, GlobalScoring *globalScoring,
+                        ScoringPerVolume *scoringPerVolume)
+{
+  TransportElectrons</*IsElectron*/ false>(positrons, active, secondaries, activeQueue, globalScoring,
+                                           scoringPerVolume);
+}
+
+template<bool IsElectron, int ProcessIndex>
+__global__
+void ComputeInteraction(Track *electrons, const adept::MParray *active, Secondaries secondaries,
+                        adept::MParray *activeQueue, GlobalScoring *globalScoring,
+                        ScoringPerVolume *scoringPerVolume)
+{
+  constexpr unsigned int sharedSize = 12250;
+  __shared__ int candidates[sharedSize];
```

Comment:
Where is this number coming from?  The size of candidates is 49KiB, which seems like an odd size to me.